main.go

```diff
@@ -0,0 +1,239 @@
+package main
+
+import (
+	"errors"
+	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
+	"net/http"
+	"time"
+)
+
+type message struct {
+	ID           string `json:"id"`
+	Category     string `json:"category"`
+	Title        string `json:"title"`
+	Author       string `json:"author"`
+	Message      string `json:"message"`
+	UUID         string `json:"uuid"`
+	Created      string `json:"created"`
+	Type         string `json:"type"`
+	Upvote       int32  `json:"upvote"`
+	Downvote     int32  `json:"downvote"`
+	CommentCount int32  `json:"commentCount"`
+	Deleted      bool   `json:"deleted"`
+}
+
+var messages = []message{
+	{ID: "1", Category: "h/AITA", Title: "Am I the asshole for killing my cat?", Author: "rakazirut", Message: "I killed my cat because he was a bitch. Am I wrong?", UUID: uuid.NewString(), Created: time.Now().Format(time.RFC3339Nano), Type: "message", Upvote: 30, Downvote: 1, CommentCount: 50, Deleted: true},
+	{ID: "2", Category: "h/Fishing", Title: "I lost my sturgeon", Author: "tslanda", Message: "Jumped off the line. Damn!", UUID: uuid.NewString(), Created: time.Now().Format(time.RFC3339Nano), Type: "message", Upvote: 300, Downvote: 14, CommentCount: 450, Deleted: false},
+}
+
+func getMessages(c *gin.Context) {
+	c.IndentedJSON(http.StatusOK, messages)
+}
+
+func helperGetMessageById(id string) (*message, error) {
+	for i, m := range messages {
+		if m.ID == id {
+			return &messages[i], nil
```

Comment:
Returns a pointer to the message located by ID. This way the request(s) can reference the record for whatever reason